opencog/guile/SchemeSmobAS.cc

```diff
@@ -90,8 +77,13 @@ void SchemeSmob::release_as (AtomSpace *as)
 		delete as;
 
 		// (Recursively) decrement the use count on the parent.
-		// XXX This is confusing, I'm not sure its right.
-		if (env) release_as(env);
+		// We had incremented it earlier, in `ss_new_as`, when
+		// creating it.
+		if (env and deleteable_as.end() != deleteable_as.find(env))
+		{
+			--deleteable_as[env];
+			if (0 == deleteable_as[env]) release_as(env);
```

Comment:
Am I right, that this and above operations on `deleteable_as` should also be under lock `lck`?

Reply:
Ok, I see you have fixed it in https://github.com/opencog/atomspace/pull/2073